nym-api/src/v2/mod.rs

```diff
@@ -77,39 +46,44 @@ impl ApiHttpServer {
 /// AFTER you have shut down BG tasks (or past their grace period).
 pub(crate) struct ShutdownHandles {
     task_manager: TaskManager,
-    axum_handle: AxumHandle,
+    axum_shutdown_button: ShutdownAxum,
+    /// Tokio JoinHandle for axum server's task
+    axum_join_handle: AxumJoinHandle,
 }
 
 impl ShutdownHandles {
     /// Cancellation token is given to Axum server constructor. When the token
     /// receives a shutdown signal, Axum server will shut down gracefully.
-    pub(crate) fn new(task_manager: TaskManager) -> (Self, WaitForCancellationFutureOwned) {
-        //
-        let token = CancellationToken::new();
-        (
-            Self {
-                task_manager,
-                axum_handle: AxumHandle(token.clone()),
-            },
-            token.cancelled_owned(),
-        )
-    }
-
-    pub(crate) fn task_manager(&self) -> &TaskManager {
-        &self.task_manager
+    pub(crate) fn new(
+        task_manager: TaskManager,
+        axum_server_handle: AxumJoinHandle,
+        shutdown_button: CancellationToken,
+    ) -> Self {
+        Self {
+            task_manager,
+            axum_shutdown_button: ShutdownAxum(shutdown_button.clone()),
+            axum_join_handle: axum_server_handle,
+        }
     }
 
     pub(crate) fn task_manager_mut(&mut self) -> &mut TaskManager {
         &mut self.task_manager
     }
 
-    /// Send signal to Axum server to gracefully shut down.
-    pub(crate) fn shutdown_axum(self) {
-        self.axum_handle.0.cancel()
+    /// Signal server to shut down, then return join handle to its
+    /// `tokio` task
+    ///
+    /// https://tikv.github.io/doc/tokio/task/struct.JoinHandle.html
+    #[must_use]
+    pub(crate) fn shutdown_axum(self) -> AxumJoinHandle {
+        self.axum_shutdown_button.0.cancel();
+        self.axum_join_handle
     }
 }
 
-struct AxumHandle(CancellationToken);
+struct ShutdownAxum(CancellationToken);
+
+type AxumJoinHandle = JoinHandle<Result<(), std::io::Error>>;
```

Comment:
For anything that contains `std::io::Error` you can use `std::io::Result<T>`, which makes this a bit shorter:`JoinHandle<std::io::Result<()>>`

Reply:
:heavy_check_mark: